jobs/furnishings/src/furnishings/stage_processors/post_processor.py

```diff
@@ -152,10 +87,119 @@ def get_info_by_name(name: Furnishing.FurnishingName) -> dict:
+        furnishing_group.save()
+        return furnishing_group, xml_payload
+
+    def update_furnishings_status(self, funishing_status, furnishing_group_id=None, notes=None):
+        """Update furnishing entries after processing."""
+        for furnishings in self._furnishings_dict.values():
+            for furnishing in furnishings:
+                if furnishing_group_id:
+                    furnishing.furnishing_group_id = furnishing_group_id
+                    furnishing.processed_date = datetime.utcnow()
+
+                if notes:
+                    furnishing.notes = notes
+
+                furnishing.status = funishing_status
+                furnishing.last_modified = datetime.utcnow()
+                furnishing.save()
+
+    def process(self):
+        """Postprocess to generate and upload file to external resources (BC Laws)."""
+        if not self._furnishings_dict:
+            return
+
+        # Create the XML data from the furnishings dict
+        xml_data = self._format_furnishings(self._furnishings_dict, self._processed_date.strftime('%I:%M %p'))
+        self._app.logger.debug('Formatted furnishing details presented in XML file')
+
+        # SFTP to BC Laws
+        payload = self._build_xml_data(xml_data)
+        filename = f'QP_CORP_{LegislationDatetime.format_as_legislation_date(self._processed_date)}.xml'
+
+        with self._bclaws_sftp_connection as client:
+            resp = client.putfo(
+                    fl=StringIO(payload),
+                    remotepath=(
+                        f'{self._app.config.get("BCLAWS_SFTP_STORAGE_DIRECTORY")}'
+                        f'/{filename}'
+                    )
+                )
+        self._app.logger.debug(f'Successfully uploaded {resp.st_size} bytes to BCLaws SFTP')
+
+        # Save xml payload
+        furnishing_group, _ = self._save_xml_payload(payload)
+        self._app.logger.debug('Saved XML payload')
+
+        # mark furnishing records processed
+        self.update_furnishings_status(Furnishing.FurnishingStatus.PROCESSED, furnishing_group_id=furnishing_group.id)
+        self._app.logger.debug(
+            f'Furnishing records with group id: {furnishing_group.id} marked as processed')
+
+
 def process(app: Flask, furnishings_dict: dict):
     """Run postprocess after stage run to upload files to external resources."""
     try:
         processor = PostProcessor(app, furnishings_dict)
         processor.process()
+    except AuthenticationException as err:
+        notes = 'SFTP Error: Unable to authenticate.'
+        processor.update_furnishings_status(Furnishing.FurnishingStatus.FAILED, notes=notes)
+        app.logger.error(err)
+    except (NoValidConnectionsError, IOError) as err:
+        notes = f'SFTP Error: {os.strerror(err.errno)}.'
+        processor.update_furnishings_status(Furnishing.FurnishingStatus.FAILED, notes=notes)
+        app.logger.error(err)
```

Comment:
Added some error handling to catch SFTP errors. When this happens:
- Furnishings are marked as failed.
- The error message is added to the furnishings note.
- The furnishing group and xml payload are not saved.